From 5164ed340956721927f6e0e6ff1699d9d7d25545 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Wed, 24 Aug 2022 09:59:27 +0300
Subject: [PATCH 088/102] dt-bindings: pinctrl: update documentation for SIUL2
 pinctrl driver

Update the pinctrl dt-bindings documentation with the one from Linux.

Issue: ALB-9214

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../pinctrl/nxp,s32cc-pinctrl.yaml            | 90 +++++++++++++++----
 1 file changed, 75 insertions(+), 15 deletions(-)

diff --git a/doc/device-tree-bindings/pinctrl/nxp,s32cc-pinctrl.yaml b/doc/device-tree-bindings/pinctrl/nxp,s32cc-pinctrl.yaml
index 0a66bbe6e6..6e0a847201 100644
--- a/doc/device-tree-bindings/pinctrl/nxp,s32cc-pinctrl.yaml
+++ b/doc/device-tree-bindings/pinctrl/nxp,s32cc-pinctrl.yaml
@@ -1,35 +1,95 @@
 # SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
 %YAML 1.2
 ---
-
+$id: http://devicetree.org/schemas/pinctrl/nxp,s32cc-siul2.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
 title: NXP S32 Common Chassis SIUL2 iomux controller
 
 maintainers:
   - Radu Pirea <radu-nicolae.pirea@nxp.com>
+  - Andrei Stefanescu <andrei.stefanescu@nxp.com>
 
 description: |
-  Core driver for the S32 Common Chassis pin controller.
+  Core driver for the pin controller found on S32 Common Chassis SoC.
 
 properties:
   compatible:
-    const:
-      - nxp,s32cc-siul2-pinctrl
+    oneOf:
+      - enum:
+          - nxp,s32g-siul2-pinctrl
+          - nxp,s32r45-siul2-pinctrl
+
+  reg:
+    minItems: 5
+    maxItems: 6
+    description: list of memory regions to be reserved
 
-  pins:
-    description: |
-      Two integers array, represents a group of pins mux and config
-      setting. The format is fsl,pins = <PIN_FUNC_ID CONFIG>
-      PIN_FUNC_ID - id of MSCR to be modified
-      CONFIG - configuration to be written in the MSCR/IMCR register
+  nxp,pins:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    description:
+      A list composed of [start, end] pins that correspond to each of the
+      memory regions reserved.
 
-      Even though IMCR register should be used as input register, it can be
-      set and addressed in the same way as MSCR, only instead of passing the
-      IMCR index, IMCR_IDX + 512 is passed.
+  '#pinctrl-cells':
+    const: 2
 
 required:
   - compatible
-  - pins
+  - reg
+  - nxp,pins
+  - '#pinctrl-cells'
+
+
+patternProperties:
+  '_pins$':
+    type: object
+    patternProperties:
+      '_grp[0-9]$':
+        type: object
+        allOf:
+          - $ref: pinmux-node.yaml#
+          - $ref: pincfg-node.yaml#
+        description:
+          Pinctrl node's client devices specify pin muxes using subnodes,
+          which in turn use the standard properties below.
 
 additionalProperties: false
 
-...
+examples:
+  - |
+    #include <dt-bindings/memory/s32-siul2.h>
+
+    pinctrl: siul2-pinctrl@4009c240 {
+        compatible = "nxp,s32g-siul2-pinctrl";
+        #pinctrl-cells = <2>;
+
+        reg = <MSCRS_SIUL2_0   0x198>,
+              <MSCRS_SIUL2_1_0 0x2c>,
+              <MSCRS_SIUL2_1_1 0xbc>,
+              <IMCRS_SIUL2_0   0x150>,
+              <IMCRS_SIUL2_1_0 0x45c>,
+              <IMCRS_SIUL2_1_1 0x108>;
+
+        /* MSCR range */
+        nxp,pins = <0   101
+                    112 122
+                    144 190
+        /* IMCR range */
+                    512 595
+                    631 909
+                    942 1007>;
+
+        llce_can0_pins {
+            llce_can0_grp0 {
+                pinmux = <0x2b0>;
+                input-enable;
+                slew-rate = <0x00>;
+            };
+
+            llce_can0_grp1 {
+                pinmux = <0x2c2>;
+                output-enable;
+                slew-rate = <0x00>;
+            };
+        };
+    };
-- 
2.17.1

